DEVDOCS-3406: Revise OAuth and API account articles (#1405)
bigcommerce/dev-docs: docs/api-docs/getting-started/about-our-api.md

# About Our APIs

> The BigCommerce suite of APIs allows you to create apps, automate store processes, and build headless ecommerce solutions. You can download or purchase third-party apps at our [App Marketplace](https://www.bigcommerce.com/apps/). We recommend that you explore the marketplace before you start building to get a feel for what's possible.

Author: resolved in pending push.

* Manage a customer's store account details

### Storefront API
The [Storefront API](/api-reference/cart-checkout/storefront-cart-api) allows you to manage customer carts and checkouts and order information client-side. Some use cases for this API are:
* Add an item with JavaScript to a shopper's cart from the Storefront
* Programmatically retrieve and display information to a customer about their recent order
* Update the billing address of a checkout
* Clear a customer's current cart

### GraphQL API
BigCommerce's [GraphQL API](/api-docs/storefront/graphql/graphql-storefront-api-overview) allows you to query product and customer data and store settings remotely or from a store's front end. Some use cases for this API are:
* Pull a product's data with Javascript into a Stencil theme
* Access customer data via the front end of a site
* Fetch category and brand details from a store's front end

### Customer Login API

> The [Customer Login API](/api-docs/storefront/customer-login-api) lets you programmatically log in customers to a BigCommerce storefront. Some use cases for this API are:

Reviewer: ```suggestion
The [Customer Login API](/api-docs/storefront/customer-login-api) lets you programmatically sign-in customers to a BigCommerce storefront. Some use cases for this API are:

```

| [Widgets](/api-reference/storefront/widgets-api) | Programmatically inject content into a BigCommerce theme. |
| [Wishlist](/api-reference/customer-subscribers/wishlist-api) | Wishlist API allows a developer to create and manage customer wishlists. |

## REST APIs (V2 & V3)
### Request Headers

> Server-to-Server request headers require `Accept`, `X-Auth-Token`, and `Content-Type` at a minimum.

Reviewer 2: ```suggestion
Server-to-Server requests require the `Accept`, `X-Auth-Token`, and `Content-Type` headers.

```

| `Transfer-Encoding` | `chunked` | Specifies the form of encoding used to transfer the resource. | `chunked`
| `X-Rate-Limit-Requests-Left` | number | Details how many remaining requests your client can make in the current window before being rate-limited. In this case, you would expect to be able to make 6 more requests in the next 3000 milliseconds; on the 7th request within 3000 milliseconds, you would be rate-limited and would receive an HTTP 429 response. | `16101491` |
| `X-Rate-Limit-Requests-Quota` | number | Shows how many API requests are allowed in the current window for your client. | `16101495` |
| `X-Rate-Limit-Time-Reset-Ms` | number | Shows how many milliseconds are remaining in the window. In this case, 3000 milliseconds – so, 3000 milliseconds after this request, the API quota will be refreshed. |`30000 `|
| `X-Rate-Limit-Time-Window-Ms` | number | Shows the size of your current rate-limiting window. | `9762` |

### Media types

> A media type is the format of the request or response body. BigCommerce APIs accept requests and send responses in JSON. You should encode requests using the UTF-8 character set. Other character sets may have unpredictable results.

Reviewer: ```suggestion
A media type is the format of the request or response body. BigCommerce APIs accept requests and send responses in JSON. You should encode requests using the UTF-8 character set since other character sets can have unpredictable results.

```

### Content types

### Request content type

> When performing a request that contains a body (eg. POST or PUT), the type of content you are sending needs to be specified in the `Content-Type` header.

Reviewer: Regarding i.e, and e.g --[style guide](https://intranet.bigcommerce.com/pages/viewpage.action?spaceKey=SST&title=Style+Guide+Word+List/) says we not use them but i'll make this your call.

Author: on the fence. I believe this was there before and that eg indicates enumerating existing cases -- so this should be the correct usage. I'll continue to reflect.

### Response content type

> There are two ways you can specify the type of content you would like to receive. The first method is to specify the `Accept` header. The second is to supply an extension to the resource you are requesting.

Reviewer: ```suggestion
There are two ways you can specify the type of content you would like to receive. The first method is to specify the `Accept` header. The second is to supply an extension to the resource you're requesting.

```

| `X-Rate-Limit-Requests-Quota` | number | Shows how many API requests are allowed in the current window for your client. | `16101495` |
| `X-Rate-Limit-Time-Reset-Ms` | number | Shows how many milliseconds are remaining in the window. In this case, 3000 milliseconds – so, 3000 milliseconds after this request, the API quota will be refreshed. |`30000 `|
| `X-Rate-Limit-Time-Window-Ms` | number | Shows the size of your current rate-limiting window. | `9762` |

### Media types

A media type is the format of the request or response body. BigCommerce APIs accept requests and send responses in JSON. You should encode requests using the UTF-8 character set since other character sets can have unpredictable results.

### Content types

### Request content type
When performing a request that contains a body (eg. POST or PUT), the type of content you are sending needs to be specified in the `Content-Type` header.

### Response content type

> There are two ways you can specify the type of content you would like to receive. The first method is to specify the `Accept` header. The second is to supply an extension to the resource you're requesting.

Author: addressed in forthcoming commit.
